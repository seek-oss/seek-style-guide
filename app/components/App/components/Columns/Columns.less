@import (reference) "~seek-style-guide/theme";

.columns {
  display: flex;
  .column {
    flex-basis: 0;
    flex-grow: 1;
    padding-right: @grid-gutter-width * 3;
    &:last-child {
      padding-right: 0;
    }
  }
}
