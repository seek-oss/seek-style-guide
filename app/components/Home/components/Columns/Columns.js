import styles from './Columns.less';

import React, { PropTypes } from 'react';

const renderColumn = (el, index) => (
  <div key={index} className={styles.column}>{el}</div>
);

export default function Columns({ children }) {
  return (
    <div className={styles.columns}>
      {children.map(renderColumn)}
    </div>
  );
}

Columns.propTypes = {
  children: PropTypes.array.isRequired
};
