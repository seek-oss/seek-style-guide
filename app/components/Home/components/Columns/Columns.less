@import (reference) "~seek-style-guide/theme";

.columns {
  display: flex;
  margin-bottom: @grid-row-height * 4;
  .column {
    flex-basis: 0;
    flex-grow: 1;
    padding-right: @grid-gutter-width * 3;
    &:last-child {
      padding-right: 0;
    }
  }
}
