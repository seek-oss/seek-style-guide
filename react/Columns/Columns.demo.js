import React from 'react';
import PropTypes from 'prop-types';
import {
  Columns,
  PageBlock,
  Card,
  Section,
  Text
} from 'seek-style-guide/react';
import classnames from 'classnames';
import styles from './Columns.less';

const ColumnsContainer = ({ component: DemoComponent, componentProps }) => (
  <PageBlock>
    <DemoComponent {...componentProps} />
  </PageBlock>
);
ColumnsContainer.propTypes = {
  component: PropTypes.any,
  componentProps: PropTypes.object.isRequired
};

const makeColumns = count =>
  [...new Array(count)].map((x, i) => (
    <Card key={i}>
      <Section>
        <Text heading>Column {i + 1}</Text>
        <Text>This card is currently acting as a column.</Text>
      </Section>
    </Card>
  ));

export default {
  route: '/columns',
  title: 'Columns',
  category: 'Layout',
  component: Columns,
  container: ColumnsContainer,
  block: true,
  initialProps: {
    children: makeColumns(2)
  },
  options: [
    {
      label: 'Number of Columns',
      type: 'radio',
      states: [
        {
          label: 'Two Columns',
          transformProps: props => props
        },
        {
          label: 'Three Columns',
          transformProps: props => ({
            ...props,
            children: makeColumns(3)
          })
        },
        {
          label: 'Four Columns',
          transformProps: props => ({
            ...props,
            children: makeColumns(4)
          })
        }
      ]
    },
    {
      label: 'States',
      type: 'checklist',
      states: [
        {
          label: 'Flexible',
          transformProps: props => ({
            ...props,
            flexible: true,
            className: classnames(styles.columns, styles.columns_flexible)
          })
        },
        {
          label: 'Tight',
          transformProps: props => ({
            ...props,
            tight: true,
            className: classnames(styles.columns, styles.columns_tight)
          })
        },
        {
          label: 'Reverse',
          transformProps: props => ({
            ...props,
            reverse: true,
            className: classnames(styles.columns, styles.columns_reverse)
          })
        }
      ]
    }
  ]
};
