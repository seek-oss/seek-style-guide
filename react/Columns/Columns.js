import styles from './Columns.less';
import React from 'react';
import PropTypes from 'prop-types';

import classnames from 'classnames';

const renderColumn = (el, index) => (
  <div key={index} className={styles.column}>
    {el}
  </div>
);

export default function Columns({
  children,
  tight,
  flexible,
  reverse,
  className
}) {
  return (
    <div
      className={classnames({
        [styles.columns]: true,
        [styles.columns_tight]: tight,
        [styles.columns_flexible]: flexible,
        [styles.columns_reverse]: reverse,
        [className]: className
      })}
    >
      {children.map(renderColumn)}
    </div>
  );
}

Columns.propTypes = {
  children: PropTypes.array.isRequired,
  tight: PropTypes.bool,
  flexible: PropTypes.bool,
  reverse: PropTypes.bool,
  className: PropTypes.string
};

Columns.defaultProps = {
  tight: false,
  flexible: false,
  reverse: false
};
