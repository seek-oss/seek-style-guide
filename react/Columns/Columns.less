@import (reference) "~seek-style-guide/theme";

.columns {
  @media @desktop {
    display: flex;
    &.columns_reverse {
      flex-direction: row-reverse;
    }
  }
}

.column {
  @media @desktop {
    flex-basis: 0;
    flex-grow: 1;

    min-width: 0;
    .columns_flexible & {
      min-width: auto;
    }

    padding: 0 (@gutter-width / 2);
    .columns_tight & {
      padding: 0 (@gutter-width / 4);
    }
    .columns:not(.columns_reverse) &:first-child,
    .columns_reverse &:last-child {
      padding-left: 0 !important;
    }
    .columns:not(.columns_reverse) &:last-child,
    .columns_reverse &:first-child {
      padding-right: 0 !important;
    }
  }
}
