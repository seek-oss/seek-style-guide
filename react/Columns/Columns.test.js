import React from 'react';
import { shallow } from 'enzyme';
import Columns from './Columns';

describe('Columns:', () => {
  it('should render columns', () => {
    const wrapper = shallow(
      <Columns>
        <div />
        <div />
      </Columns>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render tight columns', () => {
    const wrapper = shallow(
      <Columns tight>
        <div />
        <div />
      </Columns>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render flexible columns', () => {
    const wrapper = shallow(
      <Columns flexible>
        <div />
        <div />
      </Columns>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render columns in reverse', () => {
    const wrapper = shallow(
      <Columns reverse>
        <div id="1" />
        <div id="2" />
      </Columns>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
