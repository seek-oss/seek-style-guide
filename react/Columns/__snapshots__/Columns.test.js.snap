// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Columns: should render columns 1`] = `
<div
  className="Columns__columns"
>
  <div
    className="Columns__column"
    key="0"
  >
    <div />
  </div>
  <div
    className="Columns__column"
    key="1"
  >
    <div />
  </div>
</div>
`;

exports[`Columns: should render columns in reverse 1`] = `
<div
  className="Columns__columns Columns__columns_reverse"
>
  <div
    className="Columns__column"
    key="0"
  >
    <div
      id="1"
    />
  </div>
  <div
    className="Columns__column"
    key="1"
  >
    <div
      id="2"
    />
  </div>
</div>
`;

exports[`Columns: should render flexible columns 1`] = `
<div
  className="Columns__columns Columns__columns_flexible"
>
  <div
    className="Columns__column"
    key="0"
  >
    <div />
  </div>
  <div
    className="Columns__column"
    key="1"
  >
    <div />
  </div>
</div>
`;

exports[`Columns: should render tight columns 1`] = `
<div
  className="Columns__columns Columns__columns_tight"
>
  <div
    className="Columns__column"
    key="0"
  >
    <div />
  </div>
  <div
    className="Columns__column"
    key="1"
  >
    <div />
  </div>
</div>
`;
